hooks/pre_validations/30-el8_upgrade_postgresql.rb

```diff
@@ -0,0 +1,28 @@
+def check_postgresql_storage
```

Comment:
I think for foreman-maintain users we will need to include this same check there, otherwise users will not encounter this check and its consequences until they are in the middle of a Foreman upgrade.

Reply:
If only foreman-maintain could run `foreman-installer --only-checks`… ;)

Reply:
In theory it could, that would present us with a slight complication? The current order of operations, assuming we are on version N:

1. Run checks
2. Update repositories to N+1
3. Download packages to cache
4. Set up firewall
5. Stop services
6. Update packages from cache
7. Run installer

The order of operations would mean we do not have the installer available with this check until step 6 at which point the system is down for maintenance. If the user does not have enough space, they also are sitting on updated packages at this point which means reverting to a working state is perilous.

Reply:
> If only foreman-maintain could run `foreman-installer --only-checks`… ;)

If only foreman-installer could run foreman-maintain --only-checks.... ;)

Reply:
Luckilly, we still have https://github.com/theforeman/foreman_maintain/blob/master/definitions/checks/disk/available_space_postgresql12.rb lying around, so it just needs updates and being loaded at the right time :)

Reply:
https://github.com/theforeman/foreman_maintain/pull/809